### PR TITLE
Ease to set up multiple versions for manually installed tools

### PR DESCRIPTION
**Pull request by qiagu**
E.g.
in `tool_conf.xml`
```
<tool file="path to tool version 0.1" version="0.1"/>
<tool file="path to tool version 0.2" version="0.2"/>
```

### CODE REVIEW

File: lib/galaxy/tools/toolbox/base.py [hunk 1]
```diff
@@ -568,6 +569,24 @@ def _load_tool_tag_set(self, item, panel_dict, integrated_panel_dict, tool_path,
...
+            if alter_version and not tool.guid:
+                if not tool.old_id:
+                    new_id = tool.id + '/' + alter_version
+                    setattr(tool, 'old_id', tool.id)
+                    setattr(tool, 'id', new_id)
+                    setattr(tool, 'all_ids', [new_id.lower(), tool.old_id.lower()])
+                else:
+                    new_id = tool.old_id + '/' + alter_version
+                    setattr(tool, 'id', new_id)
+                    setattr(tool, 'all_ids', [new_id.lower(), tool.old_id.lower()])
```
Comment (nsoranzo) [inline lib/galaxy/tools/toolbox/base.py:588]: I think you can simplify lines 580-588:
```python
                if not tool.old_id:
                    new_id = tool.id + '/' + alter_version
                    setattr(tool, 'old_id', tool.id)
                else:
                    new_id = tool.old_id + '/' + alter_version
                setattr(tool, 'id', new_id)
                setattr(tool, 'all_ids', [new_id.lower(), tool.old_id.lower()])
```

**Comment by martenson** [comment]
I am unclear on what is this trying to do. The following should work:

```xml
    <tool file="sameid_tool_v1.xml" />
    <tool file="sameid_tool_v2.xml" /> 
```

edit: you need to rebase for the tests to pass

**Comment by mvdbeek** [comment]
You can also list different tool paths in your tool conf, as long as they have different tool version they will load and be available. This seems like the more correct way to manage this.

**Comment by qiagu** [comment]
Really!

**Pull request closed without merging**